### PR TITLE
Switch to using UUIDs

### LINKED ISSUE #154: use UUIDs for users, folders and objects IDs
#### Description

Currently users, folders and objects IDs use hex token to identify them, but these don’t guarantee uniqueness
Make use of UUIDs 

#### DoD (Definition of Done)

Generate IDs for users and objects in UUID format

#### Testing

Unit tests

### PR DESCRIPTION
**Pull request by csc-jm**
### Description

PR changes the previous randomly generated hex tokens to UUIDs instead when assigning unique IDs to users, folders and objects.

### Related issues

Fixes #154 

### Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Changes Made

- changed the id generation methods
- updated the previous unit tests

### Testing

- [x] Unit Tests

### CODE REVIEW

File: metadata_backend/api/operators.py [hunk 1]
```diff
@@ -443,9 +442,9 @@ def _generate_accession_id(self) -> str:
 
         Will be replaced later with external id generator.
         """
-        sequence = "".join(secrets.choice(string.digits) for i in range(16))
+        sequence = "".join(str(uuid.uuid4()))
```
Comment (csc-jm) [inline metadata_backend/api/operators.py:445]: Generating UUID here will result in the standard format that would look for example like this:
`123e4567-e89b-12d3-a456-777888999000`

If we don't want the hyphens, you can get a version if the UUID without the hyphens by replacing the line with this:
`sequence = "".join(uuid.uuid4().hex)`

**Comment by blankdots** [comment]
it is weird in integration tests it wants to connect to `localhost:300`
can we base this on `develop` instead of `feature/data-mirroring` ? that branch has some outstanding issues

File: metadata_backend/api/operators.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 """Operators for handling database-related operations."""
 import re
-import secrets
-import string
+import uuid
```
Comment (blankdots) [inline metadata_backend/api/operators.py:3]:
```suggestion
from uuid import uuid4
```

and it will make the syntax simpler

**Comment by csc-jm** [comment]
Oh yea, my bad. Meant to PR this to develop

File: metadata_backend/api/operators.py (hunk 1 above)
Comment (blankdots) [reply metadata_backend/api/operators.py:445]: even simpler is `uuid4().hex` it generates string by default https://docs.python.org/3/library/uuid.html#uuid.UUID.hex

Comment (blankdots) [reply metadata_backend/api/operators.py:445]: and yes please use `.hex`

**Review by blankdots — CHANGES_REQUESTED** [review]

File: tests/test_operators.py [hunk 3]
```diff
@@ -59,8 +60,8 @@ async def async_patch():
 
         MagicMock.__await__ = lambda x: async_patch().__await__()
         self.client = MagicMock()
-        self.accession_id = "EGA123456"
-        self.folder_id = "FOL12345678"
+        self.accession_id = str(uuid.uuid4())
```
Comment (blankdots) [inline tests/test_operators.py:63]:
```suggestion
        self.accession_id = uuid4().hex
```

File: tests/test_operators.py [hunk 4]
```diff
@@ -59,8 +60,8 @@ async def async_patch():
...
+        self.folder_id = str(uuid.uuid4())
```
Comment (blankdots) [inline tests/test_operators.py:64]:
```suggestion
        self.folder_id = uuid4().hex
```

File: tests/test_operators.py [hunk 5]
```diff
@@ -2,6 +2,7 @@
 import datetime
 import re
 import unittest
+import uuid
```
Comment (blankdots) [inline tests/test_operators.py:5]:
```suggestion
from uuid import uuid4
```

**Review by blankdots — CHANGES_REQUESTED** [review]
keeping tests consistent as well

**Review by blankdots — APPROVED** [review]

**Pull request merged**